CARDS-2471: Conditional improvements

marta- opened the pull request:
**Includes:**
* [CARDS-2471](https://phenotips.atlassian.net/browse/CARDS-2471) - _Conditional improvements_
* [CARDS-2480](https://phenotips.atlassian.net/browse/CARDS-2480) - _Backend conditional section evaluation fails for Long values_

[CARDS-2471]: https://phenotips.atlassian.net/browse/CARDS-2471?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CARDS-2480]: https://phenotips.atlassian.net/browse/CARDS-2480?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

marta- committed:
CARDS-2471: Conditional improvements

marta- committed:
CARDS-2471: Conditional improvements
* Added support for single selection in QuestionnaireAutocomplete
* Made the question autocomplete in ConditionalValue single select

sdumitriu commented on modules/data-entry/src/main/frontend/src/questionnaire/ConditionalSingle.jsx:
```diff
@@ -20,112 +20,85 @@
+  }
+  if (["<", ">", "<=", ">="]. includes(operation)) {
+    // These operations are not fit for arrays
+    if (a.length > 1 || b.length > 1) return false;
```
So we can't have a multivalued number answer compared to a number? I thought that could be useful. Maybe another time.

sdumitriu committed:
CARDS-2471: Conditional improvements
Backend changes

Also fixes:
CARDS-2480: Backend conditional section evaluation fails for Long values

sdumitriu committed:
CARDS-2471: Conditional improvements
Update the import scripts to not use the requireAll property on operands

sdumitriu committed:
CARDS-2471: Conditional improvements
Quick update of the cards4kids questionnaire to not use the requireAll property on operands

sdumitriu committed:
[misc] Fix broken test questionnaires

sdumitriu committed:
CARDS-2471: Conditional improvements
Update the test questionnaires to match the new rules, and add sample conditions for `includes` and `excludes`

sdumitriu reviewed (approved):
All looks good to me.

acrowthe reviewed (approved):
Looks good

sashaandjic reviewed (approved):
tested
